pkg/stacks/truncate/truncate.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package truncate provides functions for truncating Kubernetes values in a
+// predictable way offering mildly collision safe values usable in
+// deterministic field searches.
+package truncate
+
+import (
+	"crypto/sha1" // nolint:blacklist
```

Comment:
add a comment for why we're skipping the blacklist lint check here?

Reply:
Will do. It is also worth pointing out that I think our golangci-lint config may not include the blacklist linter, so this `nolint` is for other environments where this linter is included.

I've noticed warnings from `make` that this linter is not known.

Reply:
`WARN [runner/nolint] Found unknown linters in //nolint directives: blacklist `

Reply:
not sure if i fully follow this, if our `make` shows a warning for this linter, and our golangci-lint config doesn't use this linter, should we just remove it? that makes more sense to me, instead of living with a new warning in our default environment.

Reply:
I tried removing that `nolint` directive but then `gosec` fired.  I've replaced `blacklist` with `gosec` here. `make` has been appeased.